corehq/apps/translations/app_translations/upload_form.py

```diff
@@ -97,15 +106,18 @@ def _get_template_translation_el(self):
         #
         # Currently operating under the assumption that every xForm has at least
         # one translation element, that each translation element has a text node
-        # for each question and that each text node has a value node under it
-        template_translation_el = None
-        # Get a translation element to be used as a template for new elements
-        for lang in self.langs:
+        # for each question and that each text node has a value node under it.
+        # Get a translation element to be used as a template for new elements, preferably of default lang
+        default_lang = self.app.default_language
+        default_trans_el = self.itext.find("./{f}translation[@lang='%s']" % default_lang)
+        if default_trans_el.exists():
+            return default_trans_el
+        all_langs = self.app.langs
```

Comment:
This will make `all_langs` a reference to `self.app.langs` (because `self.app.langs` is a list not a tuple), so when you call `remove()` on the next line, it will remove `default_lang` from `self.app.langs` too. You probably want to use `all_langs = copy(self.app.langs)`, or skip `default_lang` instead of removing it.

Reply:
yikes... thanks for catching that.